inst/eDISH_app/modules/renderSettings/util/createControl.R

```diff
@@ -0,0 +1,87 @@
+createControl <- function(key, metadata, data, settings, ns){
+  
+  sm_key <- filter(metadata, text_key==key)
+  
+  tt_msg <- paste0("tt_msg_", key)
+  msg <- paste0("msg_", key)   
+  
+  ### get metadata for the input
+  setting_key <- as.list(strsplit(key,"\\-\\-"))
+  setting_value <- safetyGraphics:::getSettingValue(key=setting_key, settings=settings)
+  setting_label <- createSettingLabel(key)
+  setting_description <- getSettingsMetadata(text_keys=key, cols="description")
+  
+  ### if a field-level input, get metadata about the parent column-level input
+  field_column <- NULL
+  field_column_label <- NULL
+  if (!is.null(sm_key$field_column_key)){
+    field_column <- safetyGraphics:::getSettingValue(key=list(sm_key$field_column_key), settings=settings)
+    field_column_label <- getSettingsMetadata(text_key = sm_key$field_column_key, cols = "label")
+  }
+  
+  
+  ### get the choices for the selectors
```

Comment:
This is ok probably for now, but it might be cleaner to refactor to deal with the options and defaults separately at some point. Basically first set the options, and then select the default value (if any).

Reply:
@jwildfire good idea!